src/qibo/quantum_info/random_ensembles.py

```diff
@@ -0,0 +1,636 @@
+from functools import reduce
+
+import numpy as np
+
+from qibo import gates
+from qibo.config import MAX_ITERATIONS, PRECISION_TOL, raise_error
+from qibo.models import Circuit
+from qibo.quantum_info.utils import NUM_CLIFFORDS, ONEQUBIT_CLIFFORD_PARAMS
+
+
+def random_gaussian_matrix(
+    dims: int,
+    rank: int = None,
+    mean: float = None,
+    stddev: float = None,
+    seed=None,
+):
+    """Generates a random Gaussian Matrix.
+
+    Gaussian matrices are matrices where each entry is
+    sampled from a Gaussian probability distribution
+
+    .. math::
+        p(x) = \\frac{1}{\\sqrt{2 \\, \\pi} \\, \\sigma} \\, \\exp{\\left(-\\frac{(x - \\mu)^{2}}{2\\,\\sigma^{2}}\\right)}
+
+    with mean :math:`\\mu` and standard deviation :mGeneratorath:`\\sigma`.
+
+    Args:
+        dims (int): dimension of the matrix.
+        rank (int, optional): rank of the matrix. If ``None``, then ``rank == dims``. Default: ``None``.
+        mean (float, optional): mean of the Gaussian distribution.
+        stddev (float, optional): standard deviation of the Gaussian distribution.
+        seed (int or ``numpy.random.Generator``, optional): Either a generator of random numbers
+            or a fixed seed to initialize a generator. If ``None``, initializes a generator with
+            a random seed. Default: ``None``.
+
+    Returns:
+        (ndarray): Random Gaussian matrix with dimensions ``(dims, rank)``.
+
+    """
+
+    if dims <= 0:
+        raise_error(ValueError, f"dims must be type int and positive.")
+
+    if rank is None:
+        rank = dims
+    else:
+        if rank > dims:
+            raise_error(
+                ValueError, f"rank ({rank}) cannot be greater than dims ({dims})."
+            )
+        elif rank <= 0:
+            raise_error(ValueError, f"rank ({rank}) must be an int between 1 and dims.")
+
+    if stddev is not None and stddev <= 0.0:
+        raise_error(ValueError, f"stddev must be a positive float.")
+
+    if (
+        seed is not None
+        and not isinstance(seed, int)
+        and not isinstance(seed, np.random.Generator)
+    ):
+        raise_error(
+            TypeError, f"seed must be either type int or numpy.random.Generator."
+        )
+
+    local_state = (
+        np.random.default_rng(seed) if seed is None or isinstance(seed, int) else seed
+    )
+
+    if mean is None:
+        mean = 0
+    if stddev is None:
+        stddev = 1
```

Comment:
I think it is simpler if you use these values as the defaults directly in the function definition, instead of using `None` and then changing the value. You may want to include the default values in the docstring, eg. `mean (float): mean of the Gaussian distribution (default is 1).`

Reply:
thanks, done